### PR TITLE
Core: change default to strictmode=true

### LINKED ISSUE #1665: Document strictmode
Strictmode has a large impact on the system's behavior. Currently the only documentation consists of a few config flags stating that something is not allowed in strictmode, and this line in the docs:

> For production it is recommended to enable strictmode which blocks some of the unsafe configuration options (e.g. using the IRMA demo scheme).

### LINKED ISSUE #1701: Core: enable strict mode by default
Makes it safer.

- [ ] Update documentation #1796 
- [ ] Update nuts-network-local https://github.com/nuts-foundation/nuts-network-local/pull/32
- [ ] Update nuts-workshops https://github.com/nuts-foundation/nuts-workshops/pull/38
- [ ] Update nuts-go-e2e-tests https://github.com/nuts-foundation/nuts-go-e2e-test/pull/54
- [ ] Update nuts-getting-started https://github.com/nuts-foundation/nuts-getting-started/pull/1
- [ ] Update performance-test https://github.com/nuts-foundation/performance-test/pull/6

(use https://github.com/search?q=org%3Anuts-foundation+nutsfoundation%2Fnuts-node&type=code)

### PR DESCRIPTION
**Pull request by gerardsn**
Changes:
Default of strictmode is now true (closes #1701)
Updated all tests (including those that are going to fail after #1799 is merged)
Removed strictmode from vcr.Config since it was unused.
Documented strictmode behavior (closes #1665)

Remaining: 
What other documentation should be updated? For example, the getting started pages on read the docs currently still assume strictmode=false. Changing this to true would require specifying all other settings for the guide to work, but imo would distract from the goal of the guide.

### CODE REVIEW

**Comment by gerardsn** [comment]
e2e test require https://github.com/nuts-foundation/nuts-go-e2e-test/pull/54

**Comment by woutslakhorst** [comment]
> What other documentation should be updated? For example, the getting started pages on read the docs currently still assume strictmode=false. Changing this to true would require specifying all other settings for the guide to work, but imo would distract from the goal of the guide.

A simple solution is to be very clear in the logs at startup when something fails: IF YOU WANT TO DEVELOP, set strictmode to false.

File: docs/pages/deployment/configuration.rst [hunk 1]
```diff
@@ -60,4 +60,31 @@ the options table must be regenerated using the Makefile:
...
+Strict mode
+***********
+
+Several of the server options above allow the node to be configured in a way that is unsafe for production environments, but are convenient for testing or development.
+The node can be configured to run in strict mode (default) to prevent any insecure configurations.
+Below is a summary of the impact ``strictmode=true`` has on the node and its configuration.
+
+Save storage of any private key material requires some serious consideration.
+For this reason the ``crypto.storage`` backend must explicitly be set.
+Feel free to contact us to discus other backends.
```
Comment (woutslakhorst) [inline docs/pages/deployment/configuration.rst:74]: Not needed for documentation.

**Review by woutslakhorst — APPROVED** [review]
Just remove the "contact us" part ;)

**Comment by gerardsn** [comment]
> A simple solution is to be very clear in the logs at startup when something fails: IF YOU WANT TO DEVELOP, set strictmode to false.

Validation of the config values happens in several different locations making it harder to return this type of message only after an issue has been found. I've added strictmode=false to the getting started docs.

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**